uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/pageObjects/HomePage.java

```diff
@@ -16,6 +16,11 @@ public HomePage(WebDriver driver) {
         validatePageSource(driver, containsString("Where to?"));
     }
 
+    static public LoginPage goHome_redirectToLoginPage(WebDriver driver, String baseUrl) {
+        driver.get(baseUrl);
+        return new LoginPage(driver);
```

Comment:
The redirect is done by UAA, because we're not logged in. Would changing the method name from `goHome_redirectToLoginPage` to `goHome_redirectsToLoginPage` help any? The redirect action is a result of the get, not an explicit action taken by the test.

Reply:
I like the idea of "redirects". 

How do you know, in the context of this method, that we are not logged in? Could the method then be named something like `loggedOutHome` or maybe just `goToLoginPage`?

Why are you trying to document the 2 different levels of abstraction in the method name? So both `goHome` and `redirect...`?

Reply:
I added the result to the end of the method names as a result of an earlier code review. The return type indicates that we expect to end up on the login page, but some readers liked it better to also have that in the name, because the return type tends to be hidden in the chained calls that we typically used for these objects.

I have renamed the methods to try to indicate that the second part of the name is a result, not a direct action taken.

Reply:
> How do you know, in the context of this method, that we are not logged in?

The test case know whether it has logged in or not, so I put this responsibility on the test case rather than the page objects.

Reply:
But then you can't know that it "redirects to login page", can you?

Reply:
That's an expectation that the test will verify.